File: server/Netptune.Transfer/ViewModels/ExportJobViewModel.cs
using Netptune.Transfer.Enums;

namespace Netptune.Transfer.ViewModels;

public sealed record ExportJobViewModel
{
    public Guid PublicId { get; init; }

    public ExportJobStatus Status { get; init; }

    public string RecordType { get; init; } = null!;

    public ExportFormat Format { get; init; }

    public string? Name { get; init; }

    public string? FileName { get; init; }

    public long? RowCount { get; init; }

    public long? SizeBytes { get; init; }

    public int ProgressPercent { get; init; }

    public string? ProgressMessage { get; init; }

    public string? Error { get; init; }

    public bool HasArtefact { get; init; }

    public string? RequestedByUserId { get; init; }

    public string? RequestedByDisplayName { get; init; }

    public string? RequestedByPictureUrl { get; init; }

    public DateTime CreatedAt { get; init; }

    public DateTime? StartedAt { get; init; }

    public DateTime? CompletedAt { get; init; }

    public DateTime ExpiresAt { get; init; }
}

File: client/src/app/core/models/view-models/export-job-view-model.ts
export enum ExportJobStatus {
  pending = 0,
  running = 1,
  succeeded = 2,
  failed = 3,
  cancelled = 4,
  expired = 5,
}

export enum ExportFormat {
  csv = 0,
  tsv = 1,
  xlsx = 2,
  json = 3,
  ndjson = 4,
  archive = 5,
}

export interface ExportJobViewModel {
  publicId: string;
  status: ExportJobStatus;
  recordType: string;
  format: ExportFormat;
  name?: string;
  fileName?: string;
  rowCount?: number;
  sizeBytes?: number;
  progressPercent: number;
  progressMessage?: string;
  error?: string;
  hasArtefact: boolean;
  requestedByUserId?: string;
  requestedByDisplayName?: string;
  requestedByPictureUrl?: string;
  createdAt: string;
  startedAt?: string;
  completedAt?: string;
  expiresAt: string;
}

export interface ExportJobProgressEvent {
  publicId: string;
  status: ExportJobStatus;
  progressPercent: number;
  progressMessage?: string;
  error?: string;
}
